docs/source/examples/replicate_vicuna/simple_inference.py

```diff
@@ -22,21 +36,34 @@ def _get_parser():
 def evaluate(opt, output_filename):
     run_results = {}
     # Build the translator (along with the model)
-    engine = InferenceEngine(opt)
+    if inference_mode == "py":
+        print("Inference with py ...")
+        from onmt.inference_engine import InferenceEnginePY
+
+        engine = InferenceEnginePY(opt)
+    elif inference_mode == "ct2":
+        from onmt.inference_engine import InferenceEngineCT2
+
+        engine = InferenceEngineCT2(opt)
     engine.opt.src = DATASET
     start = time.time()
     scores, preds = engine.infer_file()
-    scores = [_score.cpu().numpy().tolist() for _score in sum(scores, [])]
+    engine.terminate()
     dur = time.time() - start
     print(f"Time to generate {len(preds)} answers: {dur}s")
+    if inference_mode == "py":
+        scores = [
+            [_score.cpu().numpy().tolist() for _score in _scores] for _scores in scores
+        ]
```

Comment:
Specific reason for not doing this in the `infer_file` method? This means the two subclasses don't have the exact same interface?

Reply:
Yes indeed, py scores are lists of tensors unlike ctranslate scores.